packages/handlersjs-http/lib/handlers/routed-http-request.handler.ts

```diff
@@ -79,34 +87,44 @@ export class RoutedHttpRequestHandler extends HttpHandler {
 
     });
 
-    const matchingRoute = match ? this.pathToRouteMap.get(match) : undefined;
+    const matchingRoutes = match ? this.pathToRouteMap.get(match) : undefined;
+
+    if (matchingRoutes && matchingRoutes.length) {
 
-    if (matchingRoute) {
+      const matchingRouteWithOperation = matchingRoutes.find((r) => {
 
-      const allowedMethods = matchingRoute.route.operations.map((op) => op.method);
-      const methodSupported = allowedMethods.includes(request.method);
+        const allowed = r.route.operations.map((op) => op.method);
 
-      if (!methodSupported) {
+        return allowed.includes(request.method);
+
+      });
+
+      const allowedMethods = matchingRoutes.flatMap((r) => r.route.operations.map((op) => op.method));
+
+      if (!matchingRouteWithOperation) {
 
         return of({ body: '', headers: { Allow: allowedMethods.join(', ') }, status: 405 });
 
       }
 
       // add parameters from requestPath to the request object
-      const parameters = this.extractParameters(matchingRoute.route.path.split('/').slice(1), pathSegments);
+      const parameters = this.extractParameters(matchingRouteWithOperation.route.path.split('/').slice(1), pathSegments);
       const requestWithParams = Object.assign(request, { parameters });
-      const newContext = { request: requestWithParams, route: matchingRoute.route };
-      const preResponseHandler = matchingRoute.controller.preResponseHandler;
-
-      return (preResponseHandler ? preResponseHandler.handle(newContext) : of(newContext)).pipe(
-        switchMap((preresponse) => matchingRoute.route.handler.handle(preresponse)),
+      const newContext = { request: requestWithParams, route: matchingRouteWithOperation.route };
+      const preResponseHandler = matchingRouteWithOperation.controller.preResponseHandler;
+
+      return (preResponseHandler
+        ? preResponseHandler.handle(newContext)
+        : of(newContext)
+      ).pipe(
+        switchMap((preresponse) => matchingRouteWithOperation.route.handler.handle(preresponse)),
         map((response) => ({
-          ... response,
+          ...response,
           headers: {
-            ... response.headers,
-            ... (request.method === 'OPTIONS') && { Allow: allowedMethods.join(', ') },
+            ...response.headers,
+            ...(request.method === 'OPTIONS') && { Allow: allowedMethods.join(', ') },
           },
-        })),
```

Comment:
I'm actually in favor of having a space after the triple dots, since sticky dots on a condition seem to suggest it is the condition itself which is being destructured (but no biggy).